internal/controller/purge_controller.go

```diff
@@ -111,6 +112,13 @@ func (r *PurgeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 	return ctrl.Result{}, nil
 }
 
+func noSuchHostCheck(err error, logger logr.Logger) bool {
+	lgr := func(msg string) {
+		logger.V(log.InfoLevel).Info(msg)
```

Comment:
why not just do logf.FromContext(ctx).V(log.InfoLevel).Info(msg)?

Reply:
Also maybe just DebugLevel not InfoLevel in this case

Reply:
I don't know if Debug level logs are enabled.

Reply:
As for the first comment:
I don't have access to `ctx` object here. I could pass it as an argument, but then would that be simpler? My impl. has fewer characters to type than your proposal ;)